audinterface/core/segment.py

```diff
@@ -325,24 +325,24 @@ def process_files(
         .. _audformat: https://audeering.github.io/audformat/data-format.html
 
         """
-        series = self.process.process_files(
+        y = self.process.process_files(
             files,
             starts=starts,
             ends=ends,
             root=root,
         )
-        if len(series) == 0:
+        if len(y) == 0:
             return audformat.filewise_index()
-        objs = []
-        for idx, ((file, start, _), index) in enumerate(series.items()):
-            objs.append(
-                audformat.segmented_index(
-                    files=[file] * len(index),
-                    starts=index.levels[0] + start,
-                    ends=index.levels[1] + start,
-                )
-            )
-        return audformat.utils.union(objs)
+
+        files = []
+        starts = []
+        ends = []
+        for (file, start, _), index in y.items():
+            files.extend([file] * len(index))
+            starts.extend(index.levels[0] + start)
+            ends.extend(index.levels[1] + start)
+
+        return audformat.segmented_index(files, starts, ends)
```

Comment:
It might be that we want to handle this case with some custom error message, but at the moment I would say we just assume that the segment object does not return identical segments.